pkg/inject/virtualgateway_envoy.go

```diff
@@ -15,7 +15,8 @@ const envoyVirtualGatewayEnvMap = `
   "APPMESH_VIRTUAL_NODE_NAME": "mesh/{{ .MeshName }}/virtualGateway/{{ .VirtualGatewayName }}",
   "APPMESH_PREVIEW": "{{ .Preview }}",
   "ENVOY_LOG_LEVEL": "{{ .LogLevel }}",
-  "AWS_REGION": "{{ .AWSRegion }}"
+  "AWS_REGION": "{{ .AWSRegion }}"{{ if .EnableXrayTracing }},
+  "ENABLE_ENVOY_XRAY_TRACING": "1"{{ end }}
```

Comment:
Tricky with the if templating here, haven't seen that before. 😄

Reply:
+1, I agree but going with the existing [model](https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/inject/envoy.go#L30) for now. Can definitely be done differently